File: src/test/java/io/bloco/faker/components/BoolTest.java
package io.bloco.faker.components;

import org.junit.Before;
import org.junit.Test;

import io.bloco.faker.Faker;

import static junit.framework.TestCase.assertEquals;

public class BoolTest {

    private Faker faker;

    @Before
    public void setUp() throws Exception {
        faker = new Faker();
    }

    @Test
    public void bool() throws Exception {
        float trueRatio = Math.round(Math.random() * 10f) / 10f;
        int times = 1000;
        int trueCounter = 0;

        for (int i = 0; i < times; i++) {
            if (faker.bool.bool(trueRatio)) {
                trueCounter++;
            }
        }

        assertEquals(trueCounter / (float) times, trueRatio, 0.1);
    }
}

File: lib/src/test/kotlin/io/bloco/faker/components/BoolTest.kt
package io.bloco.faker.components

import io.bloco.faker.Faker
import junit.framework.TestCase
import org.junit.Before
import org.junit.Test
import kotlin.math.roundToInt

class BoolTest {
    private lateinit var faker: Faker
    @Before
    fun setUp() {
        faker = Faker()
    }

    @Test
    fun bool() {
        val trueRatio = (Math.random() * 10f).roundToInt() / 10f
        val times = 1000
        var trueCounter = 0
        for (i in 0 until times) {
            if (faker.bool.bool(trueRatio)) {
                trueCounter++
            }
        }
        TestCase.assertEquals((trueCounter / times.toFloat()).toDouble(), trueRatio.toDouble(), 0.1)
    }
}